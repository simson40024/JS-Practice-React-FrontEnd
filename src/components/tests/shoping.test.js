import Shopping from "../shopping/shopping";
import { createRoot } from "react-dom/client";
import { screen, fireEvent } from "@testing-library/react";
import { act } from "@testing-library/react";
// import { act } from "react-dom/test-utils";
import * as hooks from "../../hook";

let container = null;
beforeEach(() => {
  container = document.createElement("div");
  document.body.appendChild(container);
});

afterEach(() => {
  container.remove();
  container = null;
});

const state = {
  user: {
    value: {
      name: "UserName",
      token: "UserTokken",
    },
  },
  pagination: {
    currentPage: 2,
    limitOnPage: 10,
    total: 4,
  },
  sorting: {},
  currency: {
    currency: "UAH",
    rate: 38.5,
  },
  product: {
    products: [
      { _id: "123", name: "first product", price: 30, imgUrl: "/img.ipg" },
      { _id: "134", name: "second product", price: 30, imgUrl: "/img.ipg" },
    ],
    status: "loaded",
    error: null,
  },
  cart: {},
};

const mock = jest.spyOn(hooks, "useAppSelector");
const mockDispatch = jest.spyOn(hooks, "useAppDispatch");
describe("Shopping", () => {
  it("should create Shopping", () => {
    mock.mockImplementation((selector) => selector(state));

    mockDispatch.mockReturnValue(jest.fn());
    act(() => {
      createRoot(container).render(<Shopping />);
    });
    expect(container).toMatchSnapshot();
  });

  // it("shold checked button add-to-cart", () => {
  //   mock.mockImplementation((selector) => selector(state));
  //   mockDispatch.mockReturnValue(jest.fn());

  //   // const mockAddButton = jest.spyOn(Product.prototype, "addProductToCart");
  //   const mockAddButton = jest.fn();

  //   act(() => {
  //     createRoot(container).render(
  //       <Product
  //         key="123"
  //         id="123"
  //         name="this cool t-shot"
  //         price="10"
  //         imgUrl="./test.img"
  //       />
  //     );
  //   });

  //   fireEvent.click(screen.getByRole("button", { name: "add to cart" }));
  //   expect(mockAddButton).toHaveBeenCalled();
  // });
});
