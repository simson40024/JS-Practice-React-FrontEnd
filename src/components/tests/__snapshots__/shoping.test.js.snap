// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Shopping should create Shopping 1`] = `
<div>
  <div
    class="sorting"
  >
    <div
      class="sorting-name"
    >
      by product:
      <select
        id="sort-by-product"
        name="sort-by-product"
      >
        <option
          selected=""
          value="-"
        >
          -
        </option>
        <option
          value="increase"
        >
          increase
        </option>
        <option
          value="decrease"
        >
          decrease
        </option>
      </select>
    </div>
    <div
      class="sorting-name"
    >
      by price:
      <select
        id="sort-by-price"
        name="sort-by-price"
      >
        <option
          selected=""
          value="-"
        >
          -
        </option>
        <option
          value="increase"
        >
          increase
        </option>
        <option
          value="decrease"
        >
          decrease
        </option>
      </select>
    </div>
    <div
      class="product-on-page"
    >
      amount on page:
      <select
        id=""
        name=""
      >
        <option
          selected=""
          value="10"
        >
          10
        </option>
        <option
          value="20"
        >
          20
        </option>
        <option
          value="30"
        >
          30
        </option>
      </select>
    </div>
  </div>
  <div
    class="pagination"
  >
    <button>
      &lt;
    </button>
    <input
      id="page-number"
      value="2"
    />
    <button>
      &gt;
    </button>
  </div>
  <div
    class="product-list"
  >
    <div
      class="product"
      id="123"
    >
      <img
        alt="product in the shop"
        src="/img.ipg"
        width="250px"
      />
      <p
        class="product-name"
      >
        first product
      </p>
      <p
        class="product-price"
      >
        1155
         
        UAH
      </p>
      <div
        class="product-quantity"
      >
        <button
          role="button"
        >
          -
        </button>
        <input
          readonly=""
          type="text"
          value="1"
        />
        <button
          role="button"
        >
          +
        </button>
      </div>
      <button
        class="add-to-cart"
        role="button"
      >
        add to cart
      </button>
    </div>
    <div
      class="product"
      id="134"
    >
      <img
        alt="product in the shop"
        src="/img.ipg"
        width="250px"
      />
      <p
        class="product-name"
      >
        second product
      </p>
      <p
        class="product-price"
      >
        1155
         
        UAH
      </p>
      <div
        class="product-quantity"
      >
        <button
          role="button"
        >
          -
        </button>
        <input
          readonly=""
          type="text"
          value="1"
        />
        <button
          role="button"
        >
          +
        </button>
      </div>
      <button
        class="add-to-cart"
        role="button"
      >
        add to cart
      </button>
    </div>
  </div>
  <div
    class="pagination"
  >
    <button>
      &lt;
    </button>
    <input
      id="page-number"
      value="2"
    />
    <button>
      &gt;
    </button>
  </div>
</div>
`;
